microsetta_public_api/repo/_beta_repo.py

```diff
@@ -11,31 +10,19 @@ def __init__(self, resources):
     def exists(self, sample_ids, metric):
         distance_matrix = self._get_resource(metric)
         if isinstance(sample_ids, str):
-            return sample_ids in distance_matrix
+            return sample_ids in distance_matrix.index
         else:
-            return [(id_ in distance_matrix) for id_ in sample_ids]
+            return [(id_ in distance_matrix.index) for id_ in sample_ids]
 
     def k_nearest(self, sample_id, metric, k=1):
         distance_matrix = self._get_resource(metric)
         if not self.exists(sample_id, metric):
             raise UnknownID(sample_id)
-        n_neighbors = len(distance_matrix.ids) - 1
+        n_neighbors = len(distance_matrix.columns)
         if k > n_neighbors:
             raise InvalidParameter(
-                f"k={k} is greater than the number of neighbors of the "
```

Comment:
I think it is useful to include the maximum in this error message. Can you include that?

Reply:
yes!